src/dbnode/storage/repair.go

```diff
@@ -441,3 +480,194 @@ func (r repairerNoOp) Start()        {}
+					zap.String("namespace", nsCtx.ID.String()),
+					zap.Time("start", start),
+					zap.Time("end", end),
+					zap.String("series", seriesID.String()),
+					zap.Error(peerSeriesIter.Err()),
+				)
+				foundMismatch = true
+				break
+			}
+
+			localDP, localUnit, localAnnotation := localSeriesIter.Current()
+			peerDP, peerUnit, peerAnnotation := peerSeriesIter.Current()
+
+			if !localDP.Timestamp.Equal(peerDP.Timestamp) {
+				r.logger.Error(
+					"timestamps did not match",
+					zap.Int("index", i),
+					zap.Time("local", localDP.Timestamp),
+					zap.Time("peer", peerDP.Timestamp),
+				)
+				foundMismatch = true
+				break
+			}
+
+			if localDP.Value != peerDP.Value {
+				r.logger.Error(
+					"Values did not match",
+					zap.Int("index", i),
+					zap.Float64("local", localDP.Value),
+					zap.Float64("peer", peerDP.Value),
+				)
+				foundMismatch = true
+				break
+			}
+
+			if localUnit != peerUnit {
+				r.logger.Error(
+					"Values did not match",
+					zap.Int("index", i),
+					zap.Int("local", int(localUnit)),
+					zap.Int("peer", int(peerUnit)),
+				)
+				foundMismatch = true
+				break
+			}
+
+			if nsCtx.Schema == nil {
+				// Remaining shadow logic is proto-specific.
+				continue
+			}
+
+			err = localM.Unmarshal(localAnnotation)
+			if err != nil {
+				r.logger.Error(
+					"Unable to unmarshal local annotation",
+					zap.Int("index", i),
+					zap.Error(err),
+				)
+				foundMismatch = true
+				break
+			}
+
+			err = peerM.Unmarshal(peerAnnotation)
+			if err != nil {
+				r.logger.Error(
+					"Unable to unmarshal peer annotation",
+					zap.Int("index", i),
+					zap.Error(err),
+				)
+				foundMismatch = true
+				break
+			}
+
+			if !dynamic.Equal(localM, peerM) {
+				r.logger.Error(
+					"Local message does not equal peer message",
+					zap.Int("index", i),
+					zap.String("local", localM.String()),
+					zap.String("peer", peerM.String()),
+				)
+				foundMismatch = true
+				break
+			}
+
+			if !bytes.Equal(localAnnotation, peerAnnotation) {
+				r.logger.Error(
+					"Local message equals peer message, but annotations do not match",
+					zap.Int("index", i),
+					zap.String("local", string(localAnnotation)),
+					zap.String("peer", string(peerAnnotation)),
+				)
+				foundMismatch = true
+				break
+			}
+
+			i++
+		}
+
+		if localSeriesIter.Err() != nil {
+			r.logger.Error(
+				"Local series iterator experienced an error",
+				zap.String("namespace", nsCtx.ID.String()),
+				zap.Time("start", start),
+				zap.Time("end", end),
+				zap.String("series", seriesID.String()),
+				zap.Int("numDPs", i),
+				zap.Error(localSeriesIter.Err()),
+			)
+		} else if foundMismatch {
+			r.logger.Error(
+				"Found mismatch between series",
+				zap.String("namespace", nsCtx.ID.String()),
+				zap.Time("start", start),
+				zap.Time("end", end),
+				zap.String("series", seriesID.String()),
+				zap.Int("numDPs", i),
+			)
+		} else {
+			r.logger.Info(
```

Comment:
Do we want to log every single shadow comparison?

Reply:
Yeah thats what the query code does (also helps you know its working). Its also only executing against a configurable percentage so I think its ok.

 I can change this to `debug` actually. That might make it a little nicer.